code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,1135 @@
+		UnregisterSignal(enraged, COMSIG_HUMAN_EARLY_UNARMED_ATTACK)
+		enraged.cut_overlay(hate_fire)
+	return ..()
+
+
+/* Green rune vores plasma and spews out xeno resin which lets you build xeno structure*/
+
+
+/obj/item/slimecross/warping/green
+	colour = "green"
+	runepath = /obj/effect/warped_rune/greenspace
+	drawing_time = 100
+
+
+/obj/effect/warped_rune/greenspace
+	icon_state = "xeno_rune"
+	desc = "We will build walls out of our fallen foes, they shall fear our very buildings."
+	max_cooldown = 100
+	///path of the xeno resin sheet to spawn by the rune
+	var/obj/item/stack/sheet/xeno_resin/resin
+
+
+/obj/item/stack/sheet/xeno_resin
+	name = "Resin sheets"
+	icon = 'icons/mob/alien.dmi'
+	icon_state = "nestoverlay" //literally just the xeno nest icon
+	merge_type = /obj/item/stack/sheet/xeno_resin
+
+
+/obj/effect/warped_rune/greenspace/Initialize()
+	. = ..()
+	START_PROCESSING(SSprocessing, src)
+
+
+/obj/effect/warped_rune/greenspace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		transmute_resin()
+
+
+/obj/effect/warped_rune/greenspace/proc/transmute_resin()
+	for(var/obj/item/stack/sheet/mineral/plasma/plasma_sheet in rune_turf)
+		resin = new(rune_turf)
+		plasma_sheet.amount--
+		if(plasma_sheet.amount <= 0)
+			qdel(plasma_sheet)
+
+
+//note : some of these can only be built ON resin weeds such as the resin nest.
+GLOBAL_LIST_INIT(resin_recipes, list ( \
+	new/datum/stack_recipe("Resin seed", /obj/structure/alien/weeds/node, 10, one_per_turf = 1, on_floor = 1, time = 100), \
+	new/datum/stack_recipe("Resin Wall", /obj/structure/alien/resin/wall, 3, one_per_turf = 1, on_floor = 1 , time = 50), \
+	new/datum/stack_recipe("Fake hatched egg", /obj/structure/alien/egg/burst, 2, one_per_turf = 1, on_floor = 1), \
+	new/datum/stack_recipe("Resin nest", /obj/structure/bed/nest , 5, one_per_turf = 1, on_floor = 0, time = 50), \
+	))
+
+
+/obj/item/stack/sheet/xeno_resin/get_main_recipes()
+	. = ..()
+	. += GLOB.resin_recipes
+
+
+/* pink rune, makes people slightly happier after walking on it*/
+
+
+/obj/item/slimecross/warping/pink
+	colour = "pink"
+	effect_desc = "Draws a rune that makes people happier!"
+	runepath = /obj/effect/warped_rune/pinkspace
+
+
+/obj/effect/warped_rune/pinkspace
+	desc = "Love is the only reliable source of happiness we have left. But like everything, it comes with a price."
+	icon_state = "love_rune"
+
+
+///adds the jolly mood effect along with hug sound effect.
+/obj/effect/warped_rune/pinkspace/Crossed(atom/movable/crossing)
+	. = ..()
+	if(istype(crossing,/mob/living/carbon/human))
+		playsound(rune_turf, "sound/weapons/thudswoosh.ogg", 50, TRUE)
+		SEND_SIGNAL(crossing, COMSIG_ADD_MOOD_EVENT,"jolly", /datum/mood_event/jolly)
+		crossing.visible_message(crossing, "<span class='notice'>You feel happier.</span>")
+
+
+/*Gold rune : Turn things over it to gold, completely fucks over economy */
+
+/obj/item/slimecross/warping/gold
```

Comment:
aw, you didn't go with my summoning rune idea from my last PR?

well, this rune still has the issues that it had before, including the lore issue of the gold material falling under the domain of metal slime cores, NOT gold slime cores

this rune might also run into the various problems that metalgen did

Reply:
In all honesty I'm probably going to change it as it's way too likely to fuck up the economy balance eventually, and can easily be abused with massive item stacks made from less expensive materials in general